File: chat/src/main/java/glitch/chat/object/irc/Emote.java
package glitch.chat.object.irc;

import com.google.common.collect.ImmutableRangeSet;
import glitch.api.objects.json.interfaces.IDObject;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class Emote implements IDObject<Integer> {
    private final Integer id;
    private final ImmutableRangeSet<Integer> indexRange;

    public String getEmoteUrl(Emote.Size emoteSize) {
        return String.format("http://static-cdn.jtvnw.net/emoticons/v1/%d/%s", getId(), emoteSize.value);
    }

    public enum Size {
        X1(1.0),
        X2(2.0),
        X3(3.0);

        private final double value;

        Size(double value) {
            this.value = value;
        }
    }
}

File: chat/src/main/kotlin/glitch/chat/irc/Emote.kt
package glitch.chat.irc

import glitch.api.objects.json.interfaces.IDObject

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 0.1.0
 */
data class Emote(
        override val id: Int,
        val indexRanges: Set<Pair<Int, Int>>
) : IDObject<Int> {

    fun getEmoteUrl(emoteSize: Emote.Size): String {
        return String.format("http://static-cdn.jtvnw.net/emoticons/v1/%d/%s", id, emoteSize.value)
    }

    enum class Size private constructor(internal val value: Double) {
        X1(1.0),
        X2(2.0),
        X3(3.0)
    }
}